rocAL/rocAL/include/api/rocal_api_types.h

```diff
@@ -155,7 +155,8 @@ enum RocalResizeScalingMode {
     ROCAL_SCALING_MODE_DEFAULT = 0,     // scales wrt specified size, if only resize width/height is provided the other dimension is scaled according to aspect ratio
     ROCAL_SCALING_MODE_STRETCH = 1,     // scales wrt specified size, if only resize width/height is provided the other dimension is not scaled
     ROCAL_SCALING_MODE_NOT_SMALLER = 2, // scales wrt to aspect ratio, so that resize width/height is not lesser than the specified size
-    ROCAL_SCALING_MODE_NOT_LARGER = 3   // scales wrt to aspect ratio, so that resize width/height does not exceed specified size
+    ROCAL_SCALING_MODE_NOT_LARGER = 3,   // scales wrt to aspect ratio, so that resize width/height does not exceed specified size
+    ROCAL_SCALING_MODE_MIN_MAX = 4 // scales wrt to aspect ratio, so that resize width/height does not exceed specified min and max size
```

Comment:
why is this new mode introduced?

Reply:
This mode has been introduced for mask training. MaskRCNN requires a different resize scaling logic that's not available in the existing modes, hence in order to follow that logic for mask training we have introduced this mode so the user can enable this instead of the generic resize logic